amethyst_assets/src/reload.rs

```diff
@@ -1,18 +1,115 @@
-//! MODULE DISABLED, will be reused later!
+//! Defines the `Reload` trait.
 
-use Source;
+use std::sync::Arc;
 
-pub trait Reload {
-    fn needs_reload(&self, source: &Source) -> bool;
+use {Asset, BoxedErr, Format, Source};
+
+/// The `Reload` trait provides a method which checks if an asset needs to be reloaded.
+pub trait Reload<A: Asset>: ReloadClone<A> + Send + Sync + 'static {
+    /// Checks if a reload is necessary.
+    fn needs_reload(&self) -> bool;
+    /// Returns the asset name.
+    fn name(&self) -> String;
+    /// Returns the format name.
+    fn format(&self) -> String;
```

Comment:
All formats has it's name as associated `&'static str`. So you can contain and return `&'static str`.

Reply:
I had this as `&'static str` first, but changed it back for an unknown reason. I'll try to change this later.